src/AzureClient/AzureEnvironment.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using Microsoft.Quantum.Simulation.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    internal class AzureEnvironment
+    {
+        public string ClientId { get; private set; } = string.Empty;
+        public string Authority { get; private set; } = string.Empty;
+        public List<string> Scopes { get; private set; } = new List<string>();
+        public Uri? BaseUri { get; private set; }
+
+        private AzureEnvironment()
+        {
+        }
+
+        public static AzureEnvironment Create(string environment, string subscriptionId) =>
+            string.IsNullOrEmpty(environment) ? Production()
+            : environment.Equals("dogfood", StringComparison.OrdinalIgnoreCase) ? Dogfood(subscriptionId)
+            : environment.Equals("canary", StringComparison.OrdinalIgnoreCase) ? Canary()
+            : Production();
+
+        private static AzureEnvironment Production() =>
+            new AzureEnvironment()
+            {
+                ClientId = "84ba0947-6c53-4dd2-9ca9-b3694761521b",      // QDK client ID
+                Authority = "https://login.microsoftonline.com/common",
+                Scopes = new List<string>() { "https://quantum.microsoft.com/Jobs.ReadWrite" },
+                BaseUri = new Uri("https://app-jobscheduler-prod.azurewebsites.net/"),
+            };
+
+        private static AzureEnvironment Dogfood(string subscriptionId) =>
+            new AzureEnvironment()
+            {
+                ClientId = "46a998aa-43d0-4281-9cbb-5709a507ac36",      // QDK dogfood client ID
+                Authority = GetDogfoodAuthority(subscriptionId),
+                Scopes = new List<string>() { "api://dogfood.azure-quantum/Jobs.ReadWrite" },
+                BaseUri = new Uri("https://app-jobscheduler-test.azurewebsites.net/"),
+            };
+
+        private static AzureEnvironment Canary()
+        {
+            var canary = Production();
```

Comment:
This makes me really look forward to C# 9's `with` keyword.

Reply:
I also had that thought while I was typing this. 👍